src/transformStateWithClones.js

```diff
@@ -1,13 +1,33 @@
-'use strict';
-
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
+"use strict";
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [{ ...state }];
+
+  for (let i = 0; i < actions.length; i++) {
+    const valueBeforePassing =
+      i === 0 ? { ...resultArray[i] } : { ...resultArray[i - 1] };
+
+    switch (actions[i].type) {
+      case "addProperties":
```

Comment:
The switch-case statements lack default case. It is always good to have a default case in your switch statement for catching unexpected cases.